java/src/main/java/com/adhoc/flight/QueryRunner.java

```diff
@@ -0,0 +1,254 @@
+             * Authentication
+             */
+            System.out.println("[INFO] [STEP 1]: Authenticating with the Dremio server using Arrow Flight " +
+                    "authorization header authentication.");
+            System.out.println("[INFO] Initial UserSession client properties are set as well.");
+            System.out.println("[INFO] Setting client property: routing-tag => test-routing-tag");
+            System.out.println("[INFO] Setting client property: routing-queue => Low Cost User Queries");
+
+            // Set routing-tag and routing-queue during initial authentication.
+            final Map<String, String> properties = ImmutableMap.of(
+                    "routing-tag", "test-routing-tag",
+                    "routing-queue", "Low Cost User Queries");
+            final HeaderCallOption routingCallOption = createClientProperties(properties);
+            // Authenticates FlightClient with routing properties.
+            client = createFlightClient(routingCallOption);
+            PrintUtils.prettyPrintAuthenticationSuccess(ARGUMENTS.host, ARGUMENTS.port);
+
+            /**
+             * Create demo table in $scratch
+             */
+            System.out.println("[INFO] [STEP 2]: Create a test table in $scratch.");
+            client.runQuery(CREATE_DEMO_TABLE, null);
+            System.out.println("[INFO] Created $scratch.dremio_flight_demo_table in $scratch successfully.");
+
+            /**
+             * Query demo table
+             */
+            System.out.println("[INFO] [STEP 3]: Query demo table $scrach.dremio_flight_demo_table");
+            System.out.println("[INFO] Setting client property: schema => $scratch");
+            PrintUtils.printRunQuery(SELECT_DEMO_TABLE);
+
+            // Set default schema path to "$scratch" for the next FlightRPC request.
+            final Map<String, String> schemaProperty = ImmutableMap.of(
+                    "schema", DEMO_TABLE_SCHEMA);
+            final HeaderCallOption schemaCallOption = createClientProperties(schemaProperty);
+            // Run query "select * from dremio_flight_demo_table" without schema path.
+            final List<Object[]> results = client.runQuery(SELECT_DEMO_TABLE, schemaCallOption);
+
+            /**
+             * Print Results
+             */
+            System.out.println("[INFO] [STEP 4]: Iterate through query results.");
+            PrintUtils.prettyPrintRows(results);
+            System.out.println();
+
+            /**
+             * Drop Demo Table
+             */
+            System.out.println("[INFO] [STEP 5]: Drop demo table.");
+            client.runQuery(DROP_DEMO_TABLE, schemaCallOption);
+            System.out.println("[INFO] Dropped $scratch.dremio_flight_demo_table successfully");
+        } catch (Exception ex) {
+            System.out.println("[ERROR] Exception: " + ex.getMessage());
+            ex.printStackTrace();
+        } finally {
+            AutoCloseables.close(client);
```

Comment:
Seems like we should use try-with-resources instead.

Reply:
Addressed